plugins/processors/cumulative_sum/cumulative_sum.go

```diff
@@ -0,0 +1,122 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package cumulative_sum
+
+import (
+	_ "embed"
+	"fmt"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type CumulativeSum struct {
+	Fields            []string        `toml:"fields"`
+	DropOriginalField bool            `toml:"drop_original_field"`
+	CleanUpInterval   config.Duration `toml:"clean_up_interval"`
+	Log               telegraf.Logger `toml:"-"`
+
+	fieldFilter filter.Filter
+	cache       map[uint64]aggregate
+	nextCleanUp time.Time
+}
+
+type aggregate struct {
+	name       string
+	tags       map[string]string
+	fields     map[string]float64
+	expireTime time.Time
+}
+
+var timeNow = time.Now
+
+func NewCumulativeSum() *CumulativeSum {
+	return &CumulativeSum{
+		DropOriginalField: true,
+		CleanUpInterval:   config.Duration(10 * time.Minute),
+		cache:             make(map[uint64]aggregate),
+	}
+}
+
+func (*CumulativeSum) SampleConfig() string {
+	return sampleConfig
+}
+
+func (c *CumulativeSum) Init() error {
+	c.nextCleanUp = timeNow().Add(time.Duration(c.CleanUpInterval))
+	if c.Fields != nil {
+		fieldFilter, err := filter.Compile(c.Fields)
+		if err != nil {
+			return fmt.Errorf("failed to create new field filter: %w", err)
+		}
+		c.fieldFilter = fieldFilter
+	}
+	return nil
+}
```

Comment:
Please use a length check here. Given your comment in the `sample.conf` I think you mean

```suggestion
func (c *CumulativeSum) Init() error {
	// Set defaults
	if len(c.Fields) == 0 {
		c.Fields = []string{"*"}
	}
	f, err := filter.Compile(c.Fields)
	if err != nil {
		return fmt.Errorf("failed to create new field filter: %w", err)
	}
	c.accept = f
	}
	
	// Create cache
	c.cache = make(map[uint64]telegraf.Metric)
	
	// Remember to reset the metrics if an interval is specified
	if c.ResetInterval > 0 {
		c.nextReset = timeNow().Add(time.Duration(c.ResetInterval))
	}

	return nil
}
```